### ISSUE TITLE
Java Enums do not produce the correct type

### ISSUE THREAD

**Issue by ChristianKernDev** [smallrye/smallrye-open-api#1973]
For the following enum example the produced Schema is not correct, as the enum type is declared as "string", but the expected values are of type "integer"

```
public enum ExampleEnum {
	_1_ONE(1, "One"),
	_2_TWO(2, "Two"),
	_3_THREE(3, "Three"),
	;

	private final Integer value;
	private final String label;

	ExampleEnum(final Integer value, final String label) {
		this.value = value;
		this.label = label;
	}

	@JsonValue
	public Integer getValue() {
		return value;
	}

	public String getLabel() {
		return label;
	}
}
```

Produced Schema:
```
      "ExampleEnum" : {
        "enum" : [ 1, 2, 3 ],
        "type" : "string"
      },
```

**Comment by MikeEdgar** [comment]
This will be fixed in #1974 . Until then, you can try using the annotation to force the type:
```java
@Schema(type = SchemaType.INTEGER, format = "int32")
public enum ExampleEnum {
	_1_ONE(1, "One"),
	_2_TWO(2, "Two"),
	_3_THREE(3, "Three"),
	;

	private final Integer value;
	private final String label;

	ExampleEnum(final Integer value, final String label) {
		this.value = value;
		this.label = label;
	}

	@JsonValue
	public Integer getValue() {
		return value;
	}

	public String getLabel() {
		return label;
	}
}
```

**Comment by ChristianKernDev** [comment]
Thank you very much for the fast fix!
